File: solutions/lua/secret-handshake/1/secret-handshake.lua
return function(n)
	local sigs = {}
	if ((n & 1) ~= 0) then table.insert(sigs, "wink") end
	if ((n & 2) ~= 0) then table.insert(sigs, "double blink") end
	if ((n & 4) ~= 0) then table.insert(sigs, "close your eyes") end
	if ((n & 8) ~= 0) then table.insert(sigs, "jump") end
	if ((n & 16) ~= 0) then
		local reversed = {}
		for i = #sigs, 1, -1 do
			table.insert(reversed, sigs[i])
		end
		return reversed
	end
	return sigs
end

File: solutions/c/secret-handshake/1/secret_handshake.c
#include "secret_handshake.h"

#include <stdlib.h>

static void rev(const char *arr[], int n) {
  int l = 0, r = n - 1;
  while (l < r) {
    const char* temp = arr[l];
    arr[l] = arr[r];
    arr[r] = temp;
    l++;
    r--;
  }
}

const char **commands(size_t number) {
	const char **sigs = malloc(sizeof(char const*) * 4);
	int sz = 0;
	if ((number & 1) != 0) sigs[sz++] = "wink";
	if ((number & 2) != 0) sigs[sz++] = "double blink";
	if ((number & 4) != 0) sigs[sz++] = "close your eyes";
	if ((number & 8) != 0) sigs[sz++] = "jump";
	if ((number & 16) != 0)
		rev(sigs, sz);
	return sigs;
}
